packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -13,35 +10,40 @@ import 'base/file_system.dart';
 import 'base/logger.dart';
 import 'base/os.dart';
 import 'base/utils.dart';
-import 'build_system/targets/icon_tree_shaker.dart';
 import 'convert.dart';
 import 'globals_null_migrated.dart' as globals;
 
+/// Whether icon font subsetting is enabled by default.
+const bool kIconTreeShakerEnabledDefault = true;
+
 /// Information about a build to be performed or used.
 class BuildInfo {
   const BuildInfo(
     this.mode,
     this.flavor, {
     this.trackWidgetCreation = false,
-    this.extraFrontEndOptions,
-    this.extraGenSnapshotOptions,
+    List<String>? extraFrontEndOptions,
+    List<String>? extraGenSnapshotOptions,
     this.fileSystemRoots,
     this.fileSystemScheme,
     this.buildNumber,
     this.buildName,
     this.splitDebugInfoPath,
     this.dartObfuscation = false,
-    this.dartDefines = const <String>[],
+    List<String>? dartDefines,
     this.bundleSkSLPath,
-    this.dartExperiments = const <String>[],
-    @required this.treeShakeIcons,
+    List<String>? dartExperiments,
+    required this.treeShakeIcons,
     this.performanceMeasurementFile,
     this.packagesPath = '.packages', // TODO(jonahwilliams): make this required and remove the default.
     this.nullSafetyMode = NullSafetyMode.sound,
     this.codeSizeDirectory,
     this.androidGradleDaemon = true,
     this.packageConfig = PackageConfig.empty,
-  });
+  }) : extraFrontEndOptions = extraFrontEndOptions ?? const <String>[],
+       extraGenSnapshotOptions = extraGenSnapshotOptions ?? const <String>[],
+       dartDefines = dartDefines ?? const <String>[],
+       dartExperiments = dartExperiments ?? const <String>[];
```

Comment:
I need to keep a look out for this pattern as I migrate, otherwise `dartDefines` or `dartExperiments` could have been passed in as null with mixed safety.